### PR TITLE
expression: using pingcap/log and unify the log format

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Unify log format in package `expression`.

### What is changed and how it works?

Replace log package and follow the new format.

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: expression/builtin_string.go [hunk 1]
```diff
@@ -264,7 +265,7 @@ func (c *concatFunctionClass) getFunction(ctx sessionctx.Context, args []Express
 
 		if argType.Flen < 0 {
 			bf.tp.Flen = mysql.MaxBlobWidth
-			log.Warningf("Not Expected: `Flen` of arg[%v] in CONCAT is -1.", i)
+			log.Warn("There's unexpected `Flen` value(-1) in CONCAT's args", zap.Int("arg's index in CONCAT", i))
```
Comment (eurekaka) [inline expression/builtin_string.go]:
```suggestion
			log.Warn("Unexpected `Flen` value(-1) in CONCAT's args", zap.Int("arg's index", i))
```

File: expression/builtin_string.go [hunk 2]
```diff
@@ -1770,7 +1771,7 @@ func getFlen4LpadAndRpad(ctx sessionctx.Context, arg Expression) int {
 	if constant, ok := arg.(*Constant); ok {
 		length, isNull, err := constant.EvalInt(ctx, chunk.Row{})
 		if err != nil {
-			log.Errorf("getFlen4LpadAndRpad with error: %v", err.Error())
+			log.Error("There's error when eval the `Flen` of LAPD/RPAD", zap.String("error", err.Error()))
```
Comment (eurekaka) [inline expression/builtin_string.go]:
```suggestion
			log.Error("Eval `Flen` for LAPD/RPAD", zap.String("error", err.Error()))
```

File: expression/builtin_string.go [hunk 3]
```diff
@@ -2150,7 +2151,10 @@ func (b *builtinCharSig) evalString(row chunk.Row) (string, bool, error) {
 	oldStr := result
 	result, _, err = transform.String(encoding.NewDecoder(), result)
 	if err != nil {
-		log.Errorf("Convert %s to %s with error: %v", oldStr, charsetName, err.Error())
+		log.Warn("Cannot convert string to target charset",
```
Comment (eurekaka) [inline expression/builtin_string.go]:
```suggestion
		log.Warn("Change charset of string",
```

File: expression/builtin_string.go [hunk 4]
```diff
@@ -2150,7 +2151,10 @@ func (b *builtinCharSig) evalString(row chunk.Row) (string, bool, error) {
...
+			zap.String("original string", oldStr),
```
Comment (eurekaka) [inline expression/builtin_string.go]:
```suggestion
			zap.String("string", oldStr),
```

File: expression/builtin_string.go [hunk 5]
```diff
@@ -2150,7 +2151,10 @@ func (b *builtinCharSig) evalString(row chunk.Row) (string, bool, error) {
...
+			zap.String("target charset", charsetName),
```
Comment (eurekaka) [inline expression/builtin_string.go]:
```suggestion
			zap.String("charset", charsetName),
```

File: expression/builtin_string.go [hunk 6]
```diff
@@ -322,7 +323,7 @@ func (c *concatWSFunctionClass) getFunction(ctx sessionctx.Context, args []Expre
 		if i != 0 {
 			if argType.Flen < 0 {
 				bf.tp.Flen = mysql.MaxBlobWidth
-				log.Warningf("Not Expected: `Flen` of arg[%v] in CONCAT_WS is -1.", i)
+				log.Warn("There's unexpected `Flen` value(-1) in CONCAT_WS's args", zap.Int("arg's index in CONCAT_WS", i))
```
Comment (zz-jason) [inline expression/builtin_string.go]:
```suggestion
				log.Warn("Unexpected `Flen` value(-1) in CONCAT_WS's args", zap.Int("arg's index", i))
```

File: expression/builtin_time.go [hunk 7]
```diff
@@ -5274,7 +5275,7 @@ func (b *builtinTimeFormatSig) evalString(row chunk.Row) (string, bool, error) {
 	dur, isNull, err := b.args[0].EvalDuration(b.ctx, row)
 	// if err != nil, then dur is ZeroDuration, outputs 00:00:00 in this case which follows the behavior of mysql.
 	if err != nil {
-		log.Warnf("Expression.EvalDuration() in time_format() failed, due to %s", err.Error())
+		log.Warn("Expression.EvalDuration in TIME_FORMAT failed", zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/builtin_time.go]: how about:
```suggestion
		log.Warn("TIME_FORMAT.args[0].EvalDuration() failed", zap.String("error", err.Error()))
```

File: expression/constant.go [hunk 8]
```diff
@@ -60,7 +61,7 @@ func (c *Constant) String() string {
 	if c.DeferredExpr != nil {
 		dt, err := c.Eval(chunk.Row{})
 		if err != nil {
-			log.Errorf("Fail to eval constant, err: %s", err.Error())
+			log.Error("Fail to eval constant to string", zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/constant.go]:
```suggestion
			log.Error("Eval constant to string", zap.String("error", err.Error()))
```

File: expression/constant_fold.go [hunk 9]
```diff
@@ -44,7 +45,7 @@ func ifFoldHandler(expr *ScalarFunction) (Expression, bool) {
 			// Failed to fold this expr to a constant, print the DEBUG log and
 			// return the original expression to let the error to be evaluated
 			// again, in that time, the error is returned to the client.
-			log.Debugf("fold constant %s: %s", expr.ExplainInfo(), err.Error())
+			log.Debug("There's error in constant folding", zap.String("expression", expr.ExplainInfo()), zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/constant_fold.go]:
```suggestion
			log.Debug("Fold expression to constant", zap.String("expression", expr.ExplainInfo()), zap.String("error", err.Error()))
```

File: expression/constant_fold.go [hunk 10]
```diff
@@ -135,7 +136,7 @@ func foldConstant(expr Expression) (Expression, bool) {
 		}
 		value, err := x.Eval(chunk.Row{})
 		if err != nil {
-			log.Warnf("fold constant %s: %s", x.ExplainInfo(), err.Error())
+			log.Debug("There's error in constant folding", zap.String("expression", x.ExplainInfo()), zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/constant_fold.go]: ditto

File: expression/constant_fold.go [hunk 11]
```diff
@@ -146,7 +147,7 @@ func foldConstant(expr Expression) (Expression, bool) {
 		if x.DeferredExpr != nil {
 			value, err := x.DeferredExpr.Eval(chunk.Row{})
 			if err != nil {
-				log.Warnf("fold constant %s: %s", x.ExplainInfo(), err.Error())
+				log.Debug("There's error in constant folding", zap.String("expression", x.ExplainInfo()), zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/constant_fold.go]: ditto

File: expression/constant_propagation.go [hunk 12]
```diff
@@ -280,7 +281,7 @@ func (s *propConstSolver) solve(conditions []Expression) []Expression {
 		s.insertCol(col)
 	}
 	if len(s.columns) > MaxPropagateColsCnt {
-		log.Warnf("[const_propagation]Too many columns in a single CNF: the column count is %d, the max count is %d.", len(s.columns), MaxPropagateColsCnt)
+		log.Warn("[const_propagation]Too many columns in a single CNF", zap.Int("Count we got", len(s.columns)), zap.Int("Expected max count", MaxPropagateColsCnt))
```
Comment (zz-jason) [inline expression/constant_propagation.go]:
```suggestion
		log.Warn("[const_propagation]Too many columns in a single CNF", zap.Int("numCols", len(s.columns)), zap.Int("maxNumCols", MaxPropagateColsCnt))
```

File: expression/constant_propagation.go [hunk 13]
```diff
@@ -528,7 +529,7 @@ func (s *propOuterJoinConstSolver) solve(joinConds, filterConds []Expression) ([
 		s.insertCol(col)
 	}
 	if len(s.columns) > MaxPropagateColsCnt {
-		log.Warnf("[const_propagation_over_outerjoin] Too many columns: column count is %d, max count is %d.", len(s.columns), MaxPropagateColsCnt)
+		log.Warn("[const_propagation_over_outerjoin]Too many columns", zap.Int("Columns we got", len(s.columns)), zap.Int("expected max count", MaxPropagateColsCnt))
```
Comment (zz-jason) [inline expression/constant_propagation.go]:
```suggestion
		log.Warn("[const_propagation_over_outerjoin]Too many columns", zap.Int("numCols", len(s.columns)), zap.Int("maxNumCols", MaxPropagateColsCnt))
```

File: expression/constraint_propagation.go [hunk 14]
```diff
@@ -150,7 +151,7 @@ func ruleConstantFalse(ctx sessionctx.Context, i, j int, exprs *exprSet) {
 	if cons, ok := cond.(*Constant); ok {
 		v, isNull, err := cons.EvalInt(ctx, chunk.Row{})
 		if err != nil {
-			log.Error(err)
+			log.Warn("[constraint_propagation]Error occurred", zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/constraint_propagation.go]:
```suggestion
			log.Warn("[constraint_propagation]ruleConstantFalse", zap.String("error", err.Error()))
```

File: expression/constraint_propagation.go [hunk 15]
```diff
@@ -247,7 +248,7 @@ func ruleColumnOPConst(ctx sessionctx.Context, i, j int, exprs *exprSet) {
 		var err error
 		fc1, err = NewFunction(ctx, scalarFunc.FuncName.L, scalarFunc.RetType, con1)
 		if err != nil {
-			log.Warn(err)
+			log.Warn("[constraint_propagation]Error occurred", zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/constraint_propagation.go]:
```suggestion
			log.Warn("[constraint_propagation]ruleColumnOPConst", zap.String("error", err.Error()))
```

File: expression/constraint_propagation.go [hunk 16]
```diff
@@ -270,7 +271,7 @@ func ruleColumnOPConst(ctx sessionctx.Context, i, j int, exprs *exprSet) {
 	}
 	v, isNull, err := compareConstant(ctx, negOP(OP2), fc1, con2)
 	if err != nil {
-		log.Warn(err)
+		log.Warn("[constraint_propagation]Error occurred", zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/constraint_propagation.go]:
```suggestion
		log.Warn("[constraint_propagation]ruleColumnOPConst", zap.String("error", err.Error()))
```

File: expression/distsql_builtin_test.go [hunk 17]
```diff
@@ -167,15 +167,15 @@ func datumExpr(d types.Datum) *tipb.Expr {
 		var err error
 		expr.Val, err = codec.EncodeDecimal(nil, d.GetMysqlDecimal(), d.Length(), d.Frac())
 		if err != nil {
-			log.Warnf("err happened when EncodeDecimal in datumExpr:%s", err.Error())
+			panic(fmt.Sprintf("err happened when EncodeDecimal in datumExpr:%s", err.Error()))
```
Comment (zz-jason) [inline expression/distsql_builtin_test.go]: how about converting it to `Assert(err, NotNil)`?

File: expression/expr_to_pb.go [hunk 18]
```diff
@@ -148,7 +149,7 @@ func (pc *PbConverter) encodeDatum(d types.Datum) (tipb.ExprType, []byte, bool)
 		var err error
 		val, err = codec.EncodeDecimal(nil, d.GetMysqlDecimal(), d.Length(), d.Frac())
 		if err != nil {
-			log.Errorf("Fail to encode value, err: %s", err.Error())
+			log.Error("[expr_to_pb]Fail to encode value", zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/expr_to_pb.go]:
```suggestion
			log.Error("[expr_to_pb]encode decimal", zap.String("error", err.Error()))
```

File: expression/expr_to_pb.go [hunk 19]
```diff
@@ -100,7 +101,7 @@ func (pc PbConverter) conOrCorColToPBExpr(expr Expression) *tipb.Expr {
 	ft := expr.GetType()
 	d, err := expr.Eval(chunk.Row{})
 	if err != nil {
-		log.Errorf("Fail to eval constant, err: %s", err.Error())
+		log.Error("[expr_to_pb]Fail to eval constant", zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/expr_to_pb.go]:
```suggestion
		log.Error("[expr_to_pb]eval constant or correlated column", zap.String("error", err.Error()), zap.String("expr", fmt.Sprintf("%s", expr.ExplainInfo())))
```

File: expression/expr_to_pb.go [hunk 20]
```diff
@@ -157,7 +158,7 @@ func (pc *PbConverter) encodeDatum(d types.Datum) (tipb.ExprType, []byte, bool)
 			t := d.GetMysqlTime()
 			v, err := t.ToPackedUint()
 			if err != nil {
-				log.Errorf("Fail to encode value, err: %s", err.Error())
+				log.Error("[expr_to_pb]Fail to encode value", zap.String("error", err.Error()))
```
Comment (zz-jason) [inline expression/expr_to_pb.go]:
```suggestion
				log.Error("[expr_to_pb]encode mysql time", zap.String("error", err.Error()))
```

**Pull request merged**